Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -710,6 +710,36 @@ CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int
     return nil;
 }
 
+size_t _CGImageImputeBitsPerPixelFromFormat(CGColorSpaceRef colorSpace, size_t bitsPerComponent, CGBitmapInfo bitmapInfo) {
+    unsigned int alphaInfo = bitmapInfo & kCGBitmapAlphaInfoMask;
+    unsigned int byteOrder = bitmapInfo & kCGBitmapByteOrderMask;
+
+    // Try byte order first: The user can specify 32 or 16 directly.
+    switch (byteOrder) {
+        case kCGBitmapByteOrder32Little:
+        case kCGBitmapByteOrder32Big:
+            return 32;
+        case kCGBitmapByteOrder16Little:
+        case kCGBitmapByteOrder16Big:
+            return 16;
+    }
+
+    // Otherwise, try to figure out how many components there are.
+    size_t nComponents = CGColorSpaceGetNumberOfComponents(colorSpace);
+    switch (alphaInfo) {
+        case kCGImageAlphaNoneSkipFirst:
+        case kCGImageAlphaPremultipliedFirst:
+        case kCGImageAlphaFirst:
+        case kCGImageAlphaNoneSkipLast:
+        case kCGImageAlphaPremultipliedLast:
+        case kCGImageAlphaLast:
+            nComponents += 1;
+            break;
+    }
+
+    return (bitsPerComponent * nComponents);
```

Comment:
This _can_ return something weird like 20 (5bpp + alpha) or 15 (5bpp no alpha) when it should return 16 for both cases.

Seeing as we don't support 16bpp contexts, however ...